File: src/main/java/ru/ifmo/is/mfl/reviews/dto/ReviewWithoutUserDto.java
package ru.ifmo.is.mfl.reviews.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.movies.dto.MovieDto;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class ReviewWithoutUserDto extends CrudDto {
  private int id;
  private MovieDto movie;
  private boolean visible;
  private String text;
  private int rating;
  private Instant date;
  private int viewedCounter;
}

File: client/src/interfaces/review/dto/ReviewWithoutUserDto.ts
import { CrudDto } from '@/interfaces/crud/dto/CrudDto';
import { MovieDto } from '@/interfaces/movie/dto/MovieDto';

export interface ReviewWithoutUserDto extends CrudDto {
  movie: MovieDto;
  visible: boolean;
  text: string;
  rating: number; // 1-10
  date: Date;
  viewedCounter: number;
}
